hazelcast/src/test/java/com/hazelcast/internal/serialization/impl/compact/CompactNullablePrimitiveInteroperabilityTest.java

```diff
@@ -175,6 +219,57 @@ public void testWriteNullReadPrimitiveThrowsException() {
         assertReadNullAsPrimitiveThrowsException(serializedRecord);
     }
 
+    @Test
+    public void testWriteNullReadPrimitiveThrowsExceptionWithCorrectMethodPrefixCompactReader() {
+        SerializationService serializationService = createSerializationServiceWithASerializer();
+        SerializationService serializationService2 = createSerializationService();
+
+        // Reading null value with non-nullable reader method
+        A a1 = new A(null, new Integer[]{1, 2, 3});
+        Data data = serializationService.toData(a1);
+        // Reading compact with serializer case
+        assertThatErrorMessageIncludes(assertThrows(HazelcastSerializationException.class, () -> serializationService.toObject(data)), "Use readNullable");
+
+        GenericRecord genericRecord = serializationService2.toObject(data);
+        // Reading compact without serializer case
+        assertThatErrorMessageIncludes(assertThrows(HazelcastSerializationException.class, () -> genericRecord.getInt32("age")), "Use getNullable");
+
+        // Reading array field with null value
+        A a2 = new A(1, new Integer[]{1, null, 3});
+        Data data2 = serializationService.toData(a2);
+        // Reading compact with serializer case
+        assertThatErrorMessageIncludes(assertThrows(HazelcastSerializationException.class, () -> serializationService.toObject(data2)), "Use readArrayOfNullable");
+
+        GenericRecord genericRecord2 = serializationService2.toObject(data2);
+        // Reading compact without serializer case
+        assertThatErrorMessageIncludes(assertThrows(HazelcastSerializationException.class, () -> genericRecord2.getArrayOfInt32("ids")), "Use getArrayOfNullable");
+    }
+
+    public void testWriteNullReadPrimitiveThrowsExceptionWithCorrectMethodPrefixGenericRecord() {
```

Comment:
how is this test different than the ones you did in the test above? we are testing the behavior of `genericRecord#getXXX` methods, and IMO they are doing the same thing in both of them. 

I would remove the checks in the `testWriteNullReadPrimitiveThrowsExceptionWithCorrectMethodPrefixCompactReader` method above regarding the generic records, and only have it here, in this test